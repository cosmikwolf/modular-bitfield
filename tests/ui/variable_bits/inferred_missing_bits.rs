use modular_bitfield::prelude::*;

#[derive(Specifier)]
#[variable_bits]  // Inferred but missing #[bits] on data variant
enum TestData {
    #[discriminant = 0]
    #[bits = 8]
    Small(u8),
    #[discriminant = 1]
    // Missing #[bits = N] for inferred mode
    Medium(u16),
}

fn main() {}