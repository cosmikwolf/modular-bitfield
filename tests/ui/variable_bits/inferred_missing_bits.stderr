error: data variant Medium requires #[bits = N] when using inferred variable_bits
  --> tests/ui/variable_bits/inferred_missing_bits.rs:9:5
   |
9  | /     #[discriminant = 1]
10 | |     // Missing #[bits = N] for inferred mode
11 | |     Medium(u16),
   | |_______________^
